src/libsyncengine/update_detection/file_system_observer/localfilesystemobserverworker.cpp

```diff
@@ -629,24 +630,24 @@ ExitCode LocalFileSystemObserverWorker::exploreDir(const SyncPath &absoluteParen
     // Process all files
     try {
         std::error_code ec;
-        for (auto dirIt = std::filesystem::recursive_directory_iterator(
-                 absoluteParentDirPath, std::filesystem::directory_options::skip_permission_denied, ec);
-             dirIt != std::filesystem::recursive_directory_iterator(); ++dirIt) {
-            if (ec) {
-                LOG_SYNCPAL_DEBUG(_logger, "Error in exploreDir " << ec.message().c_str());
-                continue;
-            }
-
+        auto dirIt = std::filesystem::recursive_directory_iterator(
```

Comment:
This is the main change of this pull-request: abort recursion if the directory iterator construction issued an error.